src/collectors/windows.plugin/perflib-hyperv.c

```diff
@@ -1353,11 +1412,158 @@ static bool do_hyperv_network_adapter(PERF_DATA_BLOCK *pDataBlock, int update_ev
+            p->rd_PacketsSentSec = rrddim_add(p->st_send_receive_packets, "sent", NULL, -1, 1, RRD_ALGORITHM_INCREMENTAL);
+
+            rrdlabels_add(p->st_send_receive_packets->rrdlabels, "vm_net_interface", windows_shared_buffer, RRDLABEL_SRC_AUTO);
+
+            p->st_send_receive_bytes = rrdset_create_localhost(
+              "vm_net_interface_traffic",
+              windows_shared_buffer,
+              NULL,
+              HYPERV,
+              HYPERV ".vm_net_interface_traffic",
+              "VM interface traffic",
+              "kilobits/s",
+              _COMMON_PLUGIN_NAME,
+              _COMMON_PLUGIN_MODULE_NAME,
+              chart_priority++,
+              update_every,
+              RRDSET_TYPE_AREA);
+
+            p->rd_BytesReceivedSec = rrddim_add(p->st_send_receive_bytes, "received", NULL, 8, 1000, RRD_ALGORITHM_INCREMENTAL);
+            p->rd_BytesSentSec = rrddim_add(p->st_send_receive_bytes, "sent", NULL, -8, 1000, RRD_ALGORITHM_INCREMENTAL);
+            rrdlabels_add(p->st_send_receive_bytes->rrdlabels, "vm_net_interface", windows_shared_buffer, RRDLABEL_SRC_AUTO);
+
+            p->st_IPsecoffloadBytes = rrdset_create_localhost(
+                "vm_net_interface_ipsec_traffic",
+                windows_shared_buffer,
+                NULL,
+                HYPERV,
+                HYPERV ".vm_net_interface_ipsec_traffic",
+                "VM interface traffic",
+                "kilobits/s",
+                _COMMON_PLUGIN_NAME,
+                _COMMON_PLUGIN_MODULE_NAME,
+                chart_priority++,
+                update_every,
+                RRDSET_TYPE_AREA);
+
+            p->rd_IPsecoffloadBytesReceivedSec =
+                rrddim_add(p->st_IPsecoffloadBytes, "received", NULL, 8, 1000, RRD_ALGORITHM_INCREMENTAL);
+            p->rd_IPsecoffloadBytesSentSec =
+                rrddim_add(p->st_IPsecoffloadBytes, "sent", NULL, -8, 1000, RRD_ALGORITHM_INCREMENTAL);
+            rrdlabels_add(
+                p->st_IPsecoffloadBytes->rrdlabels, "vm_net_interface", windows_shared_buffer, RRDLABEL_SRC_AUTO);
+
+            p->st_DirectedPackets = rrdset_create_localhost(
+                "vm_net_interface_directed_packets",
+                windows_shared_buffer,
+                NULL,
+                HYPERV,
+                HYPERV ".vm_net_interface_directed_packets",
+                "VM interface traffic",
+                "packets/s",
+                _COMMON_PLUGIN_NAME,
+                _COMMON_PLUGIN_MODULE_NAME,
+                chart_priority++,
+                update_every,
+                RRDSET_TYPE_LINE);
+
+            p->rd_DirectedPacketsSentSec =
+                rrddim_add(p->st_DirectedPackets, "received", NULL, 1, 1, RRD_ALGORITHM_INCREMENTAL);
+            p->rd_DirectedPacketsReceivedSec =
+                rrddim_add(p->st_DirectedPackets, "sent", NULL, -1, 1, RRD_ALGORITHM_INCREMENTAL);
```

Comment:
`p->rd_DirectedPacketsSentSec` and `p->rd_DirectedPacketsReceivedSec` are swapped